clients/java/signalr/src/test/java/HubConnectionTest.java

```diff
@@ -369,15 +395,39 @@ public void SendWithFiveParamsTriggersOnHandler() throws Exception {
         assertEquals(12, value5.get(), 0);
     }
 
-    private class MockEchoTransport implements Transport {
+    @Test
+    public void ReceiveHandshakeResponseAndMessage() throws Exception {
+        AtomicReference<Double> value = new AtomicReference<Double>(0.0);
+        MockTransport transport = new MockTransport();
+        HubConnection hubConnection = new HubConnection("http://example.com", transport);
+
+        hubConnection.on("inc", () ->{
+            assertEquals(0.0, value.get(), 0);
+            value.getAndUpdate((val) -> val + 1);
+        });
+
+        // On start we're going to receive the handshake response and also an invocation in the same payload.
+        hubConnection.start();
+
+        String expectedSentMessage  = "{\"protocol\":\"json\",\"version\":1}" + RECORD_SEPARATOR;
+        assertEquals(expectedSentMessage, transport.getSentMessages()[0]);
+
+        transport.receiveMessage("{}" + RECORD_SEPARATOR +"{\"type\":1,\"target\":\"inc\",\"arguments\":[]}" + RECORD_SEPARATOR);
```

Comment:
nit: `+` spacing (the second one)

definitely seems like an auto-formatter would save a bunch of time here ;P